lib/spack/docs/containers.rst

```diff
@@ -591,6 +592,37 @@ the example below:
    COPY data /share/myapp/data
    {% endblock %}
 
+The template also defines a number of inner blocks for finer-grained control over the rendered Dockerfile:
+
+* `make_manifest`
+* `install_software`
+* `strip_binaries`
+* `make_activate`
+* `build_stage_extra_instructions`
+* `copy_into_final`
+* `make_entrypoint`
+* `final_stage_extra_instructions`
+* `set_entrypoint`
```

Comment:
If we add inner blocks, I would change the names slightly:
```suggestion
* `manifest`
* `install_software`
* `strip_binaries`
* `activation_script`
```
Also, the blocks:
- copy_into_final
- make_manifest
- set_manifest

are not inner blocks of "final_stage". I still have to think about names, but they should be documented above, in case.

Reply:
Yes, and I already suggested that, and it wasn't considered safe by @wdconinc.  https://github.com/spack/spack/pull/41558.  

All the details are in the first paragraph of this PR.